pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/kit/KitDao.java

```diff
@@ -186,6 +190,30 @@ public Boolean isBloodKit(String ddpLabel) {
         return booleanCheckFoundAsName(ddpLabel, SQL_IS_BLOOD_KIT_QUERY);
     }
 
+    public String getCollaboratorParticipantId(String ddpParticipantId) {
+        if (StringUtils.isBlank(ddpParticipantId)) {
+            return null;
+        }
+
+        SimpleResult results = inTransaction(conn -> {
+            SimpleResult dbVals = new SimpleResult("");
+            try (PreparedStatement stmt = conn.prepareStatement(SQL_GET_COLLABORATOR_PARTICIPANT_ID)) {
+                stmt.setString(1, ddpParticipantId);
+                try (ResultSet rs = stmt.executeQuery()) {
+                    if (rs.next()) {
+                        dbVals.resultValue = rs.getString(DBConstants.COLLABORATOR_PARTICIPANT_ID);
```

Comment:
This code would be clearer if you removed the indirection through `SimpleResults`, and just returned a value here. (You can also add the logging from below here.) That way anything that falls through the bottom of these conditionals can be handled uniformly. 
Likewise you can either catch and rethrow exceptions below, or not handle exceptions at this level, depending on your preferred error handling approach.

Reply:
refactored and cleaned up